sydent/db/sqlitedb.py

```diff
@@ -47,3 +58,89 @@ def __init__(self, syd):
 
         c.close()
         self.db.commit()
+
+    def _upgradeSchema(self):
+        curVer = self._getSchemaVersion()
+
+        if curVer < 1:
+            cur = self.db.cursor()
+
+            # add auto_increment to the primary key of local_threepid_associations to ensure ids are never re-used,
+            # allow the mxid column to be null to represent the deletion of a binding
+            # and remove not null constraints on ts, notBefore and notAfter (again for when a binding has been deleted
+            # and these wouldn't be very meaningful)
+            logger.info("Migrating schema from v0 to v1")
+            cur.execute("DROP INDEX IF EXISTS medium_address")
+            cur.execute("DROP INDEX IF EXISTS local_threepid_medium_address")
+            cur.execute("ALTER TABLE local_threepid_associations RENAME TO old_local_threepid_associations");
+            cur.execute(
+                "CREATE TABLE local_threepid_associations (id integer primary key autoincrement, "
```

Comment:
triple-quotes are totally a thing btw

Reply:
(for reference, doesn't need changing now)

Reply:
oh true, although istr synapse used separate single line quotes. whatever.